feat(tekton)!: update tekton UX

divyanshiGupta opened the pull request:
Fixes: https://github.com/janus-idp/backstage-plugins/issues/804

This PR updates tekton plugin UX based on new designs. It also removes support for a separate tekton tab.

Screenshots:



https://github.com/janus-idp/backstage-plugins/assets/20724543/dd157912-f7d8-46fd-8b65-5d638dec6db8



To-Do:

- [x] Update Readme
- [x] Add more unit tests

divyanshiGupta commented:
cc: @ShiranHi

ShiranHi commented:
@divyanshiGupta that looks great!
I have 3 questions:
1. Is it possible to decrease the padding between the arrow used to collapse and expand rows in the table and the first column NAME? The current gap looks quite large.
2. Could we consider including tooltips for when you hover over the "Collapse all" and "Expand all" icons?
2. How about adding an icon for each status in the filter? Let's go with the same icons we use on the table

divyanshiGupta commented:
> @divyanshiGupta that looks great! I have 3 questions:
> 
> 1. Is it possible to decrease the padding between the arrow used to collapse and expand rows in the table and the first column NAME? The current gap looks quite large.
> 2. Could we consider including tooltips for when you hover over the "Collapse all" and "Expand all" icons?
> 3. How about adding an icon for each status in the filter? Let's go with the same icons we use on the table

@ShiranHi thanks for the feedback. Will try to add the changes that you have suggested.

debsmita1 commented on plugins/tekton/src/components/common/TableExpandCollapse.tsx:
```diff
@@ -0,0 +1,41 @@
+  );
+  // const [collapseDisabled, setCollapseDisabled] =
+  //   React.useState<boolean>(false);
+  // const [expandDisabled, setExpandDisabled] = React.useState<boolean>(false);
```
remove these comments

debsmita1 commented:
The PipelineRunVisualization component doesn't need its own TektonResourcesContext Provider.

debsmita1 commented:
In the dark theme mode, the list looks like this
![Screenshot 2023-10-19 at 3 39 04 PM](https://github.com/janus-idp/backstage-plugins/assets/22490998/28194f7d-cc3b-4746-967e-f4d05f7dba71)

divyanshiGupta commented:
@ShiranHi I have reduced padding for expand/collapse buttons and added tooltips as well. Also header table cells have same padding for all but the gap between name and status cell seem more because the plr name occupies more space and hence has more width so the name cell aligns itself with it to look consistent hence increasing the space between the cells.


https://github.com/janus-idp/backstage-plugins/assets/20724543/ad7b4c48-2b3e-4cea-ad8d-8e5579dbe66e


@debsmita1 I have fixed the dark theme issue.

divyanshiGupta commented:
@ShiranHi currently the MUI core `Select` component currently doesnt support anything apart from strings. There is another component  `Automcomplete` https://v4.mui.com/components/autocomplete/  from MUI lab which supports adding icons but since this component is under lab it might be experimental and is not used anywhere else as of now in the code base. cc: @invincibleJai

ShiranHi commented:
@divyanshiGupta got it. And thank you for adding the tooltips, looks good to me.

invincibleJai commented on plugins/tekton/src/components/PipelineRunList/PipelineRunListSearchBar.tsx:
```diff
@@ -0,0 +1,83 @@
+import Search from '@material-ui/icons/Search';
+
+type PipelineRunListSearchBarProps = {
+  pipelineRuns?: any[];
```
nit: can we have type for it?

```suggestion
  pipelineRuns?: PipelineRunKind[];
```

invincibleJai reviewed (commented):
Thanks @divyanshiGupta, this looks good. Have added a few nits.

invincibleJai commented:
@divyanshiGupta I see one edge case with multi-cluster and two warnings 


https://github.com/janus-idp/backstage-plugins/assets/5129024/c1a59ec5-51e0-48f5-ba53-fa7be2051a2c


<img width="1898" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/4967a051-e59a-4eea-9cca-d8ed13e54338">

<img width="1061" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/0c4024a8-f184-4fd5-94c8-cd21342f986a">

divyanshiGupta commented:
> @divyanshiGupta I see one edge case with multi-cluster and two warnings
> 
>  tektonUX2.mov 
> <img alt="image" width="1898" src="https://user-images.githubusercontent.com/5129024/277376625-4967a051-e59a-4eea-9cca-d8ed13e54338.png"> <img alt="image" width="1061" src="https://user-images.githubusercontent.com/5129024/277376719-0c4024a8-f184-4fd5-94c8-cd21342f986a.png">

This issue is not reproducible for me. 

https://github.com/janus-idp/backstage-plugins/assets/20724543/5ad87ef1-9509-4d53-b316-33c3393942f7

Also I dont see the first warning. The warning regarding the key prop is from main.

invincibleJai commented:
@divyanshiGupta Okay, let's get this and we observe this issue we can take it up an an issue.
